worker/uniter/timer.go

```diff
@@ -4,21 +4,26 @@
 package uniter
 
 import (
+	"math/rand"
 	"time"
-)
 
-const (
-	// interval at which the unit's status should be polled
-	statusPollInterval = 5 * time.Minute
+	"github.com/juju/juju/worker/uniter/remotestate"
 )
 
-// updateStatusSignal returns a time channel that fires after a given interval.
-func updateStatusSignal() <-chan time.Time {
-	// TODO(fwereade): 2016-03-17 lp:1558657
-	return time.After(statusPollInterval)
-}
+// NewUpdateStatusTimer returns a func returning timed signal suitable for update-status hook.
+func NewUpdateStatusTimer() remotestate.UpdateStatusTimerFunc {
+	r := rand.New(rand.NewSource(time.Now().Unix()))
+	return func(wait time.Duration) func() <-chan time.Time {
+		// Actual time to wait is randomised to be +/-20%
+		// of the nominal value.
+		lower := 0.8 * float64(wait)
+		window := 0.4 * float64(wait)
+		offset := float64(r.Int63n(int64(window)))
+		wait = time.Duration(lower + offset)
 
-// NewUpdateStatusTimer returns a timed signal suitable for update-status hook.
-func NewUpdateStatusTimer() func() <-chan time.Time {
-	return updateStatusSignal
+		return func() <-chan time.Time {
+			// TODO(fwereade): 2016-03-17 lp:1558657
```

Comment:
Thought about it but a lot to try and get done for 2.2.1. I can see if I get time.